### PR TITLE
Add a customlist_explain script

### PR DESCRIPTION
**Pull request by io7m**
## Description

This adds a new `customlist_explain` script that consumes the output of the `customlist_import` command and produces a CSV file including manual remediation steps required to fix any failures that may have occurred during the import.

Specifically, this:

  * Adjusts the report schema to correctly use anyOf instead of oneOf.
  * Adjusts schemas to carry identifier types everywhere
  * Adjusts schemas to carry author names everywhere
  * Adds a customlist_explain script that produces a CSV file listing
    remediation steps.

This also adds a readme file in `customlists/README.md` that give a brief overview on how to use the scripts.

## Motivation and Context

We need some means for people who are copying customlists across servers to know what manual steps they might need to perform in the face of errors.

Affects: https://www.notion.so/lyrasis/Support-manual-remediation-for-discrete-custom-list-entries-that-could-not-be-migrated-d178f85b63be4814b33a4497b245cbeb

## How Has This Been Tested?

I tested this against a synthesized report, and added a new test case to check that the command produces the expected CSV.

## Checklist

- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.

### CODE REVIEW

File: customlists/customlist_explain.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+                        problem.id(),
+                        problem.title(),
+                        problem.author(),
+                        type(problem).TYPE,
+                        problem.message(),
+                        "The book could not be retrieved on the source CM, but it might be available with a manual search "
+                        "on the target CM.".replace("\n", ""),
+                    ]
+                )
+                continue
```
Comment (RishiDiwanTT) [inline customlists/customlist_explain.py:79]: Do we need `continues` everywhere or can this be if-elif-else(raise). While adding new problem cases people are going to forget to add continues, which would be an annoying error to debug.

Comment (io7m) [reply customlists/customlist_explain.py:79]: It could be `if-elif-raise`. I'll make that change now.

Comment (io7m) [reply customlists/customlist_explain.py]: How I wish we had algebraic data types and pattern matching... :weary:

**Comment by RishiDiwanTT** [comment]
This looks fine, apart from the test failure.

**Comment by io7m** [comment]
I'm not clear on what's causing that test failure... It's occurring in code I didn't touch (even indirectly).

**Comment by RishiDiwanTT** [comment]
Not sure what magic dust you sprinkled on the github servers, but its all working now

**Comment by io7m** [comment]
I did nothing! :sweat_smile:

**Review by RishiDiwanTT — APPROVED** [review]

**Pull request merged**